### ISSUE TITLE
MagnificationCurve and lack of methods information

### ISSUE THREAD

**Issue by rpoleski** [rpoleski/MulensModel#36]
```python
>>> import MulensModel as mm
>>> params = mm.ModelParameters({'t_0': 0, 'u_0': 1., 't_E': 9., 's': 1.5, 'q': 0.5, 'alpha': 90.})
>>> mag_curve = mm.MagnificationCurve([0, 1., 10.], params)
>>> mag_curve.get_magnification()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../source/MulensModel/magnificationcurve.py", line 164, in get_magnification
    magnification = self.get_binary_lens_magnification()
  File ".../source/MulensModel/magnificationcurve.py", line 384, in get_binary_lens_magnification
    method = methods[index].lower()
AttributeError: 'NoneType' object has no attribute 'lower'
```
The message should give some useful information.

**Comment by jenniferyee** [comment]
I don't think this is a quick fix because this error occurs for binary lenses but not point lenses. Therefore, the implementation of methods is not universal, so needs to be rethought. This is related to the overall architecture problem with how we've implemented models and magnification methods. See branch [refactor_models](https://github.com/rpoleski/MulensModel/tree/refactor_models).

**Comment by rpoleski** [comment]
I agree that improving whole Model and MagnificationCurve architectures should solve this one, but I think we can add a test (`if methods[index] is None`:`) and an error message without huge changes in most of the code. 
Now the error is in line 423 of `magnificationcurve.py`.

**Comment by rpoleski** [comment]
Solved in 90c18c5
